app/assets/javascripts/redux/modules/models/promotionalTweets.js

```diff
@@ -17,6 +25,25 @@ export const initialState = {
   loading: false,
 };
 
+function handlePromoTweetSuccessResponse(state, action) {
+  const normalizedPromoTweet = normalizeListenSignalsResponse(action.payload);
```

Comment:
Well, you're normalizing a listen signal response; this a promo tweet post; the payload will be different. You don't need to normalize the response. You're only getting back one promo tweet.